### PR TITLE
test: Use consistent assert functions

### PR DESCRIPTION
**Pull request by NorrinRadd**
ref #23119: Updated a subset of the test cases to get approach feedback before making a large amount of changes that may need to change again. 

Approach:

- assert_equal is preferred over assert where possible. 
- A matching assert_not_equal function has been added also. 
- Where assert_equal is not possible, the built-in assert function will be used. 

Rationale:
This has been discussed in a few PRs / Issue.  The most recent consensus is that a [minimal approach](https://github.com/bitcoin/bitcoin/pull/25732) is best.

### CODE REVIEW

**Review by shaavan — COMMENTED** [review]
Concept ACK on using the consistent assert function.

However, I am skeptical of this statement.

> Other variations of assert will be removed.

There are many variations of assert in `test/test_framework/util.py`, and all serve different purposes. For example, `assert_is_hex_string` will be used in the case of verifying hex strings. 
If we eliminate all variants other than `assert_equal` and `assert_not_equal`, it will increase the verbosity of the assert code on the usage side, making maintaining and debugging code a challenge.

IMO, I would rather have a bunch of variations of assert statements based on a different case, maintained in the same file, and used consistently throughout the codebase than have only a minimal amount of functions implemented and use them to write custom code on the usage side.

File: .gitignore [hunk 1]
```diff
@@ -56,6 +56,7 @@ src/qt/bitcoin-qt.creator.user
 src/qt/bitcoin-qt.files
 src/qt/bitcoin-qt.includes
 
+tags
```
Comment (brunoerg) [inline .gitignore:59]: Reason for this?

Comment (NorrinRadd) [reply .gitignore:59]: This is the tags indexing file, used for quickly navigating to symbols within the project. Many IDEs use it and it updates as changes happen so it is not static.

**Comment by NorrinRadd** [comment]
> 

That is an important distinction. assert and assert_equal will be used where possible. 

> Concept ACK on using the consistent assert function.
> 
> However, I am skeptical of this statement.
> 
> > Other variations of assert will be removed.
> 
> There are many variations of assert in `test/test_framework/util.py`, and all serve different purposes. For example, `assert_is_hex_string` will be used in the case of verifying hex strings. If we eliminate all variants other than `assert_equal` and `assert_not_equal`, it will increase the verbosity of the assert code on the usage side, making maintaining and debugging code a challenge.
> 
> IMO, I would rather have a bunch of variations of assert statements based on a different case, maintained in the same file, and used consistently throughout the codebase than have only a minimal amount of functions implemented and use them to write custom code on the usage side.

Yes, that's an important point.  I'll update the summary accordingly and also revert some of the removals of assert_greater_than, as I can see in previous PRs that function was acceptable.

**Comment by NorrinRadd** [comment]
Any more issues? Is this good to merge? Once accepted I can make these changes to the other TC files and hopefully close out #23119

File: test/functional/test_framework/util.py [hunk 2]
```diff
@@ -51,9 +51,19 @@ def assert_fee_amount(fee, tx_size, feerate_BTC_kvB):
...
+            err_msg = "not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args)
+        raise AssertionError(err_msg)
+
+
+def assert_not_equal(thing1, thing2, *args, err_msg=None):
+    if thing1 == thing2 or any(thing1 == arg for arg in args):
+        msg = "%s" % " == ".join(str(arg) for arg in (thing1, thing2) + args)
+        if err_msg is not None:
+            msg = err_msg
+        raise AssertionError(msg)
```
[the remark this answers is not in the document]
Comment (NorrinRadd) [reply test/functional/test_framework/util.py:66]: This makes sense. More efficient.

**Review by aureleoules — APPROVED** [review]
LGTM. I think you should squash the commits.

**Comment by fanquake** [comment]
Moved to draft for now. Seems like there could be agreement on making this change, but it needs rebasing (2 months), changes need squashing, needs review comments addressing.

**Pull request closed without merging**